src/realm.js

```diff
@@ -206,6 +213,12 @@ export class Realm {
   $GlobalEnv: LexicalEnvironment;
   intrinsics: Intrinsics;
 
+  react: {
+    enabled: boolean,
+    componentsByName: Map<string, FunctionalComponent>,
```

Comment:
Why do you need a component by name? Won't it be enough to just keep track of a `Set` of all `FunctionValue`s that we want to optimize. Should be able to have only nameless components.

Reply:
I agree, I'll take this out for now and put it into another PR at a later point as per what you said.